doc/extensibility/adding_xaml_rules.md

```diff
@@ -0,0 +1,116 @@
+# Adding Xaml Rules
+
+In CPS, the role of XAML rules is to describe to CPS _what_ and _how_ properties, items, and metadata
+from msbuild matter. There are several ways to add XAML rules to the project for CPS to pick up.
+
```

Comment:
I think the document need explain why there are those different kinds of ways to add rules, and what developers should use.  Otherwise, it will be hard to make the right choice, when there are many options.

Reply:
I have added this section.